commit b3822a1078c87a6f74270741bc4cc660e4f11bae
Author: Eric Anholt <eric@anholt.net>
Date:   Mon Feb 15 19:03:57 2016 -0800

    clk: bcm2835: Fix setting of PLL divider clock rates
    
    commit 773b3966dd3cdaeb68e7f2edfe5656abac1dc411 upstream.
    
    Our dividers weren't being set successfully because CM_PASSWORD wasn't
    included in the register write.  It looks easier to just compute the
    divider to write ourselves than to update clk-divider for the ability
    to OR in some arbitrary bits on write.
    
    Fixes about half of the video modes on my HDMI monitor (everything
    except 720x400).
    
    Signed-off-by: Eric Anholt <eric@anholt.net>
    Signed-off-by: Michael Turquette <mturquette@baylibre.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/clk/bcm/clk-bcm2835.c b/drivers/clk/bcm/clk-bcm2835.c
index 39bf582..4f9830c 100644
--- a/drivers/clk/bcm/clk-bcm2835.c
+++ b/drivers/clk/bcm/clk-bcm2835.c
@@ -1097,13 +1097,15 @@ static int bcm2835_pll_divider_set_rate(struct clk_hw *hw,
 	struct bcm2835_pll_divider *divider = bcm2835_pll_divider_from_hw(hw);
 	struct bcm2835_cprman *cprman = divider->cprman;
 	const struct bcm2835_pll_divider_data *data = divider->data;
-	u32 cm;
-	int ret;
+	u32 cm, div, max_div = 1 << A2W_PLL_DIV_BITS;
 
-	ret = clk_divider_ops.set_rate(hw, rate, parent_rate);
-	if (ret)
-		return ret;
+	div = DIV_ROUND_UP_ULL(parent_rate, rate);
+
+	div = min(div, max_div);
+	if (div == max_div)
+		div = 0;
 
+	cprman_write(cprman, data->a2w_reg, div);
 	cm = cprman_read(cprman, data->cm_reg);
 	cprman_write(cprman, data->cm_reg, cm | data->load_mask);
 	cprman_write(cprman, data->cm_reg, cm & ~data->load_mask);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1420_b3822a1078c87a6f74270741bc4cc660e4f11bae.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1420_b3822a1078c87a6f74270741bc4cc660e4f11bae.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
